### PR TITLE
compare function finished

### PR DESCRIPTION
**Pull request by Knilchon**
compare function finished.

### CODE REVIEW

File: src/typeScript/presentation/quizPage.ts [hunk 1]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
-window.onload = function(){
+window.onload = function () {
     initilize();
     incQuestionIndex();
     //console.log(quiz);
 };
-   
-function initilize(){
+
+function initilize() {
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts]: rename in something better

File: src/typeScript/presentation/quizPage.ts [hunk 2]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
     let element = document.getElementById("wrapQuestion");      //declares connection to a given div
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts]: do fix the problem you can add a type assertion
like 
let element = document.getElementById("wrapQuestion") as HTMLInputElement;

**Review by rheien — CHANGES_REQUESTED** [review]

File: src/typeScript/presentation/quizPage.ts [hunk 3]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
 
     let nextQuestionElement = document.createElement("button");    //creates a NEXT question Element
-    nextQuestionElement.addEventListener("click",nextQuestion);
-    element.appendChild(nextQuestionElement); 
+    nextQuestionElement.addEventListener("click", nextQuestion);
+    element.appendChild(nextQuestionElement);
     nextQuestionElement.id = "nextQuestionElement";
     nextQuestionElement.textContent = "Next";
 
     let submitElement = document.createElement("button");       //creates a SUBMIT button Element
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts:64]: let submitElement = document.getElementById("submitElement") as HTMLButtonElement

File: src/typeScript/presentation/quizPage.ts (hunk 1 above)
Comment (Knilchon) [reply src/typeScript/presentation/quizPage.ts]: 💀🗿🗿🗿🗿🗿🗿🗿🗿

File: src/typeScript/presentation/quizPage.ts [hunk 4]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
 }
-function incQuestionIndex(){
-    document.getElementById("questionElement").textContent = quiz[prozess-1].Question;
-    document.getElementById("answer1Element").textContent = quiz[prozess-1].Answers[0];
-    document.getElementById("answer2Element").textContent = quiz[prozess-1].Answers[1];
-    document.getElementById("answer3Element").textContent = quiz[prozess-1].Answers[2];
-    document.getElementById("answer4Element").textContent = quiz[prozess-1].Answers[3];
+
+function incQuestionIndex() {
+    document.getElementById("questionElement").textContent = quiz[prozess - 1].Question;
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts]: let questionElement = document.getElementById("progressElement") as HTMLLabelElement;
    questionElement.textContent = quiz[prozess - 1].Question;

**Review by rheien — CHANGES_REQUESTED** [review]

File: src/typeScript/presentation/quizPage.ts (hunk 2 above)
Comment (rheien) [reply src/typeScript/presentation/quizPage.ts]: everything is an element but you can specify it with a name like renderQuestion or something else

File: src/typeScript/presentation/quizPage.ts [hunk 5]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
 
     let questionElement = document.createElement("label");       //creates a question Element
-    element.appendChild(questionElement);   
+    element.appendChild(questionElement);
     questionElement.id = "questionElement";
 
     let answer1Element = document.createElement("button");       //creates answer Elements
-    answer1Element.addEventListener("click",selectFunc);
-    element.appendChild(answer1Element); 
+    answer1Element.addEventListener("click", selectFunc);
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts:24]: why not firstAnswerButton or something else

File: src/typeScript/presentation/quizPage.ts [hunk 6]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
     answer4Element.id = "answer4Element";
     answer4Element.className = "answerElements"
 
 
     let progressElement = document.createElement("label");       //creates a PROGRESS Element
-    element.appendChild(progressElement); 
+    element.appendChild(progressElement);
     progressElement.id = "progressElement";
-    progressElement.textContent =  prozess.toString() + "/5";
+    progressElement.textContent = prozess.toString() + "/5";
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts:51]: what progress?

File: src/typeScript/presentation/quizPage.ts [hunk 7]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
     }
+
 }
-function submitFunc(){
+
+function submitFunc() {
     //hideSub Button
-        let submitElement = document.getElementById("submitElement");
-        submitElement.style.display = 'none';
+    let submitElement = document.getElementById("submitElement");
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts]: submitButton

File: src/typeScript/presentation/quizPage.ts [hunk 8]
```diff
@@ -5,110 +5,151 @@ let quiz = questionShuffle(questions);
...
-function compareAnswers(){
-    //compares button labels with given answer and tells them if they should be slected => /bool shouldSelect = true;
-    //compares if shouldSelect is true and if the value is "Selected"
 }
-function timer(){
-    
+
+function compareAnswers() {
+
+    let rightAnswers = new Array;
```
Comment (rheien) [inline src/typeScript/presentation/quizPage.ts:130]: correctAnswers?

**Review by rheien — CHANGES_REQUESTED** [review]

**Comment by rheien** [comment]
not a good style to read

**Pull request merged**